Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean

```diff
@@ -201,6 +201,14 @@ theorem fg_adjoin_of_finite {s : Set A} (hfs : s.Finite) (his : ∀ x ∈ s, IsI
           (his a <| Set.mem_insert a s).fg_adjoin_singleton)
     his
 
+theorem Algebra.finite_adjoin_of_finite_of_isIntegral {s : Set A} (hf : s.Finite)
+    (hi : ∀ x ∈ s, IsIntegral R x) : Module.Finite R (adjoin R s) :=
+  Module.Finite.iff_fg.mpr <| fg_adjoin_of_finite hf hi
+
+theorem Algebra.finite_adjoin_simple_of_isIntegral {x : A} (hi : IsIntegral R x) :
+    Module.Finite R (adjoin R {x}) :=
+  Module.Finite.iff_fg.mpr hi.fg_adjoin_singleton
```

Comment:
The proofs are short but I think these are useful since they removes the `toSubmodule` from `fg_adjoin_of_finite` and `fg_adjoin_singleton`.

There is [Subalgebra.finiteDimensional_toSubmodule](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/FiniteDimensional/Defs.html#Subalgebra.finiteDimensional_toSubmodule) that does this removal but we need to generalize it to `Module.Finite` and make it an alias.